src/app/(main)/my-voyage/[teamId]/features/components/FeaturesComponentWrapper.tsx

```diff
@@ -71,81 +65,105 @@ function transformData(features: Features[]): FeaturesList[] {
   return transformedData;
 }
 
-interface FetchFeaturesProps {
-  teamId: number;
-}
+// interface FetchFeaturesProps {
+//   teamId: number;
+// }
 
-export async function fetchFeatures({
-  teamId,
-}: FetchFeaturesProps): Promise<AsyncActionResponse<FeaturesList[]>> {
-  let data: FeaturesList[] | null = [];
-  const token = getAccessToken();
-
-  const fetchFeaturesAsync = () =>
-    GET<Features[]>(
-      `api/v1/voyages/teams/${teamId}/features`,
-      token,
-      "force-cache",
-      CacheTag.features,
-    );
+// export async function fetchFeatures({
+//   teamId,
+// }: FetchFeaturesProps): Promise<AsyncActionResponse<FeaturesList[]>> {
+//   let data: FeaturesList[] | null = [];
+//   const token = getAccessToken();
 
-  const [res, error] = await handleAsync(fetchFeaturesAsync);
+//   const fetchFeaturesAsync = () =>
+//     GET<Features[]>(
+//       `api/v1/voyages/teams/${teamId}/features`,
+//       token,
+//       "force-cache",
+//       CacheTag.features,
+//     );
 
-  if (res) {
-    data = transformData(res);
-  }
+//   const [res, error] = await handleAsync(fetchFeaturesAsync);
 
-  return [data, error];
-}
+//   if (res) {
+//     data = transformData(res);
+//   }
+
+//   return [data, error];
+// }
 
 interface FeaturesComponentWrapperProps {
   params: {
     teamId: string;
   };
 }
 
-export default async function FeaturesComponentWrapper({
+export default function FeaturesComponentWrapper({
   params,
 }: FeaturesComponentWrapperProps) {
-  let features = [];
+  // let features = [];
 
   const teamId = Number(params.teamId);
-
-  const [user, error] = await getUser();
-
-  const { errorResponse, data } = await getCurrentVoyageData({
-    user,
-    error,
-    teamId,
-    args: { teamId },
-    func: fetchFeatures,
-  });
-
-  if (errorResponse) {
-    return (
-      <ErrorComponent
-        errorType={ErrorType.FETCH_VOYAGE_DATA}
-        message={errorResponse}
-      />
-    );
-  }
-
-  if (data) {
-    const [res, error] = data;
-
-    if (error) {
-      return (
-        <ErrorComponent
-          errorType={ErrorType.FETCH_FEATURES}
-          message={error.message}
-        />
-      );
-    }
-
-    features = res!;
-  } else {
-    redirect("/");
-  }
+  const [features, setFeatures] = useState<FeaturesList[]>([]);
+
+  useEffect(() => {
```

Comment:
Could we do something like that maybe?

// Function to fetch features
const fetchFeatures = async (teamId: string, setFeatures: Function) => {
  try {
    const response = await axiosInstance.get(
      `/api/v1/voyages/teams/${teamId}/features`,
    );
    const transformedFeatures = transformData(response.data);
    setFeatures(transformedFeatures);
  } catch (error: any) {
    console.error('Failed to fetch features', error);
  }
};

  useEffect(() => {
    fetchFeatures(teamId, setFeatures); // Simply call the function
  }, [teamId]);